apps/sms/services/js/bridge_client_mixin.js

```diff
@@ -0,0 +1,144 @@
+/*global bridge */
+/*jshint esnext: true */
+
+'use strict';
+
+(function(exports) {
+
+  const priv = Object.freeze({
+    client: Symbol('client'),
+    serviceName: Symbol('serviceName'),
+    appInstanceId: Symbol('appInstanceId'),
+    pendingOperations: Symbol('pendingOperations'),
+
+    initBridge: Symbol('initBridge'),
+    registerOperation: Symbol('registerOperation')
+  });
+
+  /**
+   * This mixin allows to expose easily a service from an object.
+   *
+   * This needs the following properties to work properly:
+   *
+   * - SERVICE_NAME holds the service name.
+   */
+  var clientMixin = {
+    /**
+     * Initialize service to listen on the default context and given endpoint.
+     * @param {(Iframe|Worker|MessagePort|BroadcastChannel|Window)} endpoint
+     *  The context/thread that service could listen for.
+     */
+    initClient({ appInstanceId, endpoint, plugins }) {
+      if (!appInstanceId) {
+        throw new Error('Application instance identifier is required!');
+      }
+
+      this[priv.appInstanceId] = appInstanceId;
+      this[priv.initBridge](endpoint || exports, plugins);
+    },
+
+    clientMethod(...params) {
```

Comment:
Right now calling `.destroy()` calls `.disconnect()` which will cancel any pending calls. If you don't want this behaviour you should probably wait for the stream to close before calling `.destroy()`.